lib/expression/callbacks.ex

```diff
@@ -1214,13 +1282,30 @@ defmodule Expression.Callbacks do
       true
       iex> Expression.evaluate!("@has_date_eq(\\"there is no date here, just a year 2017\\", \\"2017-01-15\\")")
       false
+      iex> Expression.evaluate!("@has_date_eq(date(2022, 12, 12), date(2022, 12, 12))")
+      true
+      iex> Expression.evaluate!("@has_date_eq(\\"the date is 15/01/2017\\", \\"2017-01-15\\").match")
+      ~D[2017-01-15]
+
   """
   def has_date_eq(ctx, expression, date_string) do
     [expression, date_string] = eval_args!([expression, date_string], ctx)
     found_date = extract_dateish(expression)
     test_date = extract_dateish(date_string)
-    # Future match result: found_date
-    found_date == test_date
+
+    result =
+      case found_date do
+        found_date when is_struct(found_date, DateTime) ->
+          DateTime.compare(found_date, test_date) == :eq
```

Comment:
What if `test_date` is not a `DateTime` struct? Could that happen?

Reply:
actually, no, today() returns a date according to the spec.